vitest.shared.mts

```diff
@@ -0,0 +1,87 @@
+import { resolve, dirname } from 'node:path';
+import { fileURLToPath } from 'node:url';
+import { defineConfig } from 'vitest/config';
+
+const CURRENT_DIR = dirname(fileURLToPath(import.meta.url));
+const WORKSPACE_ROOT = resolve(CURRENT_DIR, './');
+
+export default defineConfig({
+  // We seem to need both this and the `env` property below to make it work.
+  define: {
+    'process.env.NODE_ENV': '"test"',
+    'process.env.VITEST': '"true"',
+    LICENSE_DISABLE_CHECK: 'false',
+  },
+  esbuild: {
+    minifyIdentifiers: false,
+    keepNames: true,
+  },
+  resolve: {
+    alias: [
+      // Generates resolver aliases for all packages and their plans.
+      ...[
+        { lib: 'x-charts', plans: ['pro'] },
+        { lib: 'x-date-pickers', plans: ['pro'] },
+        { lib: 'x-tree-view', plans: ['pro'] },
+        { lib: 'x-data-grid', plans: ['pro', 'premium', 'generator'] },
+        { lib: 'x-internals' },
+        { lib: 'x-license' },
+        { lib: 'x-telemetry' },
+      ].flatMap((v) => {
+        return [
+          {
+            find: `@mui/${v.lib}`,
+            replacement: resolve(WORKSPACE_ROOT, `./packages/${v.lib}/src`),
+          },
+          ...(v.plans ?? []).map((plan) => ({
+            find: `@mui/${v.lib}-${plan}`,
+            replacement: resolve(WORKSPACE_ROOT, `./packages/${v.lib}-${plan}/src`),
+          })),
+        ];
+      }),
+      {
+        find: 'test/utils',
+        replacement: new URL('./test/utils', import.meta.url).pathname,
+      },
+    ],
+  },
+  test: {
+    globals: true,
+    setupFiles: [new URL('test/setupVitest.ts', import.meta.url).pathname],
+    // Required for some tests that contain early returns.
+    // Should be removed once we migrate to vitest.
+    passWithNoTests: true,
```

Comment:
Is this still needed? 🤔

Reply:
Yeah, there are some suites that have dynamic/conditional tests